packages/blocks/src/api/utils.js

```diff
@@ -341,12 +342,33 @@ export function __experimentalGetBlockAttributesNamesByRole( name, role ) {
 	if ( ! role ) {
 		return attributesNames;
 	}
-	return attributesNames.filter(
-		( attributeName ) =>
-			attributes[ attributeName ]?.__experimentalRole === role
-	);
+
+	return attributesNames.filter( ( attributeName ) => {
+		const attribute = attributes[ attributeName ];
+		if ( attribute?.role === role ) {
+			return true;
+		}
+		if ( attribute?.__experimentalRole === role ) {
+			deprecated( '__experimentalRole attribute', {
+				since: '6.7',
+				version: '6.8',
+				alternative: 'role attribute',
+			} );
+			return true;
+		}
+		return false;
+	} );
 }
 
+export const __experimentalGetBlockAttributesNamesByRole = ( ...args ) => {
```

Comment:
Possibly not exported correctly from the blocks package. Will need to check the `index` file

Reply:
✅